### PR TITLE
fix: remove deprecated usage of parse

### PR DESCRIPTION
**Pull request by jbl428**
**Summary**

Replace depreacted parse function to WHATWG URL.

**Checklist**

- The basics
  - [x] I tested these changes manually in my local or dev environment
- Tests
  - [x] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [x] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [x] N/A

### CODE REVIEW

**Comment by Nezteb** [comment]
FYI @chohmann I added TC as a reviewer. This PR is old/stale, so I just wanted to make sure it's on your radar!

File: packages/http/src/__tests__/client.spec.ts [hunk 1]
```diff
@@ -42,11 +42,11 @@ describe('User Http Client', () => {
         jest.spyOn(client, 'request');
       });
 
-      afterAll(() => jest.clearAllMocks());
```
Comment (daniel-white) [inline packages/http/src/__tests__/client.spec.ts:45]: why was this moved into the inner scopes?

File: packages/http/src/client.ts [hunk 2]
```diff
@@ -30,16 +30,25 @@ export function createClientFromOperations(resources: IHttpOperation[], defaultC
...
+        O.tryCatch(() => new URL(url)),
+        O.getOrElseW(() => new URL(url, 'https://stoplight.io'))
+      );
 
       const httpUrl: IHttpUrl = {
-        baseUrl: parsedUrl.host ? `${parsedUrl.protocol}//${parsedUrl.host}` : mergedConf.baseUrl,
+        baseUrl,
         path: parsedUrl.pathname,
-        query: parseQueryString(parsedUrl.query || '') as IHttpNameValues,
+        query: parseQueryString(parsedUrl.search.substring(1)) as IHttpNameValues,
```
Comment (daniel-white) [inline packages/http/src/client.ts:51]: is `search` guaranteed not to be undefined?

File: packages/http/src/forwarder/__tests__/index.spec.ts [hunk 3]
```diff
@@ -4,11 +4,12 @@ import { assertResolvesRight, assertResolvesLeft } from '@stoplight/prism-core/s
 import { keyBy, mapValues } from 'lodash';
 import { hopByHopHeaders } from '../resources';
 import { DiagnosticSeverity, Dictionary } from '@stoplight/types';
+import { format } from 'url';
```
Comment (daniel-white) [inline packages/http/src/forwarder/__tests__/index.spec.ts:7]: i'm not seeing any functional changes here

File: packages/http/src/forwarder/index.ts [hunk 4]
```diff
@@ -41,9 +41,11 @@ const forward: IPrismComponents<IHttpOperation, IHttpRequest, IHttpResponse, IHt
       TE.chainEitherK(() => serializeBody(input.body)),
       TE.chain(body =>
         TE.tryCatch(async () => {
-          const partialUrl = parse(baseUrl);
+          const partialUrl = new URL(baseUrl);
           const url = format({
             ...partialUrl,
+            host: partialUrl.host,
```
Comment (daniel-white) [inline packages/http/src/forwarder/index.ts:47]: why include this if is already spread in from line 46?

**Review by daniel-white — CHANGES_REQUESTED** [review]

File: packages/http/src/forwarder/__tests__/index.spec.ts (hunk 3 above)
Comment (jbl428) [reply packages/http/src/forwarder/__tests__/index.spec.ts]: It is my mistake :)

File: packages/http/src/__tests__/client.spec.ts (hunk 1 above)
Comment (jbl428) [reply packages/http/src/__tests__/client.spec.ts:46]: I thought that previous mock's call history should be cleared before each `describe` started .

If second `describe` pass, the third one always pass because both  have same `test` body. 

```ts
      describe('when calling a method with overridden url', () => {
        beforeAll(() => client.get('/pet', { baseUrl: 'https://www.google.it' }));

        afterAll(() => jest.clearAllMocks());

        test('should call the mocker with the replaced full url', () => {
          expect(mock.default).toBeCalledWith({ // <-- same body
            resource: expect.anything(),
            input: expect.objectContaining({
              data: expect.objectContaining({
                url: expect.objectContaining({
                  baseUrl: 'https://www.google.it',
                  path: '/pet',
                }),
              }),
            }),
            config: expect.anything(),
          });
        });
      });

      describe('when calling a method with a full url', () => {
        beforeAll(() => client.get('https://www.google.it/pet'));

        afterAll(() => jest.clearAllMocks());

        test('should call the mocker with the replaced full url', () => {
          expect(mock.default).toBeCalledWith({  // <-- same body
            resource: expect.anything(),
            input: expect.objectContaining({
              data: expect.objectContaining({
                url: expect.objectContaining({
                  baseUrl: 'https://www.google.it',
                  path: '/pet',
                }),
              }),
            }),
            config: expect.anything(),
          });
        });
      });
    })
```

File: packages/http/src/client.ts (hunk 2 above)
Comment (jbl428) [reply packages/http/src/client.ts:51]: Yes, I found that it is always string even though there are no query string in URL. 
but I think it is better to make code more defensive 😄

File: packages/http/src/forwarder/index.ts (hunk 4 above)
Comment (jbl428) [reply packages/http/src/forwarder/index.ts]: Since `partialUrl.host` is a getter not property. It is not copied properly. 

Come to think of it, it would be better to use `Object.assign` instead of spread. :)

File: docs/guides/01-mocking.md [hunk 5]
```diff
@@ -146,7 +146,7 @@ Testing against the exact same piece of data over and over again is not the best
 
 Dynamic mode solves this by generating a random value for all the properties according to their type, and other information like `format` or even the all-powerful `x-faker` extension.
 
-All the random properties are generated using [Faker.js](https://github.com/marak/Faker.js) under the hood, via [JSON Schema Faker](https://github.com/json-schema-faker/json-schema-faker). The `x-faker` keyword is optional, but when present allows for a specific Faker API method to be used (of which there [are a lot](https://github.com/marak/Faker.js#api-methods)) so you get a lot of control over the response.
+All the random properties are generated using [Faker.js](https://fakerjs.dev) under the hood, via [JSON Schema Faker](https://github.com/json-schema-faker/json-schema-faker). The `x-faker` keyword is optional, but when present allows for a specific Faker API method to be used (of which there [are a lot](https://fakerjs.dev/guide)) so you get a lot of control over the response.
```
Comment (jbl428) [inline docs/guides/01-mocking.md:149]: I have changed the link to pass CI temporary.

**Comment by chohmann** [comment]
The code looks good! Can you resolve the docs conflicts? We'll be good to go then!

**Review by daniel-white — APPROVED** [review]

**Review by brendarearden — APPROVED** [review]
